### PR TITLE
Bump self-requirement in generated requirements.txt

### PR DESCRIPTION
**Pull request by felixfontein**
Fixes problem described in https://github.com/ansible-community/antsibull-docs/pull/125#issuecomment-1510375078.

### CODE REVIEW

File: .github/workflows/antsibull-docs.yml [hunk 1]
```diff
@@ -74,6 +74,12 @@ jobs:
           cat build.sh
         working-directory: antsibull-docs
 
+      - name: Patch requirements.txt to remove antsibull-docs requirement
+        run: |
+          sed -i requirements.txt -e 's!antsibull-docs[^a-zA-Z0-9_-].*!!g'
+          cat requirements.txt
```
Comment (gotmax23) [inline .github/workflows/antsibull-docs.yml:80]: Nit: I'd add a
```suggestion
          cat requirements.txt
          (! grep antsibull-docs requirements.txt)
```

for good measure.

File: .github/workflows/antsibull-docs.yml [hunk 2]
```diff
@@ -74,6 +74,12 @@ jobs:
           cat build.sh
         working-directory: antsibull-docs
 
+      - name: Patch requirements.txt to remove antsibull-docs requirement
+        run: |
+          sed -i requirements.txt -e 's!antsibull-docs[^a-zA-Z0-9_-].*!!g'
```
Comment (gotmax23) [inline .github/workflows/antsibull-docs.yml:79]: Nit: Why not delete the line entirely?

```suggestion
          sed -i requirements.txt -e '/antsibull-docs[^a-zA-Z0-9_-].*/d'
```

File: src/antsibull_docs/data/sphinx_init/requirements_txt.j2 [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-antsibull-docs >= 1.0.0, < 2.0.0
+antsibull-docs >= 2.0.0, < 3.0.0
```
Comment (gotmax23) [inline src/antsibull_docs/data/sphinx_init/requirements_txt.j2:5]: The current version `2.0.0a1` isn't included in this range. I assume we'll create a pre-release before 2.0.0, so we don't want to generate requirements.txt files that can't be resolved. We can change this after we release 2.0.0.

```suggestion
antsibull-docs >= 2.0.0a1, < 3.0.0
```

Comment (felixfontein) [reply src/antsibull_docs/data/sphinx_init/requirements_txt.j2]: Fixed in 42b2d95658e889f0f389494a9b2bc7712446c3a1.

File: changelogs/fragments/126-sphinx-init-self-reqs.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "The antsibull-docs requirement in the ``requirements.txt`` file created by the sphinx-init subcommand now has version range ``>= 2.0.0, < 3.0.0`` (https://github.com/ansible-community/antsibull-docs/pull/126)."
```
Comment (gotmax23) [inline changelogs/fragments/126-sphinx-init-self-reqs.yml:2]:
```suggestion
  - "The antsibull-docs requirement in the ``requirements.txt`` file created by the sphinx-init subcommand now has version range ``>= 2.0.0a1, < 3.0.0`` (https://github.com/ansible-community/antsibull-docs/pull/126)."
```

:smile:

**Comment by gotmax23** [comment]
LGTM. Thanks. Let's see what CI says.

**Comment by felixfontein** [comment]
Now with https://github.com/ansible-community/antsibull-core/pull/59 merged it should hopefully go back to green.

**Comment by felixfontein** [comment]
And it did! @gotmax23 thanks for reviewing!

**Pull request merged**